src/constants.js

```diff
@@ -1,18 +1,27 @@
-// Internal._mode bitfield constants
-/** Normal mount/patch. Not hydration or side-effecting top-level render. */
-export const MODE_NONE = 0;
+// Internal._flags bitfield constants
+export const TYPE_TEXT = 1 << 0;
+export const TYPE_ELEMENT = 1 << 1;
+export const TYPE_CLASS = 1 << 2;
+export const TYPE_FUNCTION = 1 << 3;
+export const TYPE_FRAGMENT = 1 << 4;
+
+/** Any type of component */
+export const TYPE_COMPONENT = TYPE_CLASS | TYPE_FUNCTION | TYPE_FRAGMENT;
+
+// Modes of rendering
 /** Normal hydration that attaches to a DOM tree but does not diff it. */
-export const MODE_HYDRATE = 1;
+export const MODE_HYDRATE = 1 << 5;
 /** Top level render unspecified behaviour (old replaceNode parameter to render) */
-export const MODE_MUTATIVE_HYDRATE = 2;
+export const MODE_MUTATIVE_HYDRATE = 1 << 6;
 /** Signifies this VNode suspended on the previous render */
-export const MODE_SUSPENDED = 4;
+export const MODE_SUSPENDED = 1 << 7;
 /** Signifies this VNode errored on the previous render */
-export const MODE_ERRORED = 8;
+export const MODE_ERRORED = 1 << 8;
+
 /** Signifies that bailout checks will be bypassed */
-export const FORCE_UPDATE = 16;
+export const FORCE_UPDATE = 1 << 9;
 /** Signifies that a node needs to be updated */
-export const DIRTY = 32;
+export const DIRTY_BIT = 1 << 10;
```

Comment:
Are these modes? I left them as is for now, but open to changing them at other's suggestion.